src/main/resources/com/google/api/codegen/java/build_gapic.gradle.snip

```diff
@@ -19,11 +19,22 @@
 
   dependencies {
     compile 'com.google.api:gax:{@metadata.gaxVersionBound.lower}'
-    compile 'com.google.api:gax-grpc:{@metadata.gaxGrpcVersionBound.lower}'
+    testCompile 'com.google.api:gax:{@metadata.gaxVersionBound.lower}:testlib'
```

Comment:
This is surprising. It seems that discogapic feature tries to go in parallel with gapic generation (basically having its own hierarcy of classes with unfortunately significant amount of code duplication between gapic and discogapic). For example discogapic has its own `MainDiscoGapicProviderFactory`,  but here it is trying to reuse `guilb_gapic.gradle.snip`. This is somewhat subjective, but I think it should be another way around: we try to reuse classes if we can, if we can reuse classes then we try to reuse related templates (.snip files). If we failed to reuse classes, then probably we should not even try to reuse snip files (since discogapic and gapic are already in "parallel worlds").

@garrettjonesgoogle what do you think about it?

Reply:
The snippet files are explicitly tied to ViewModel classes.  I don't think it helps to extend that coupling of snippet files with generator classes all the way up with ProviderFactory classes.

Reply:
+1  to @andreamlin - snippet files should be coupled to view model classes and not transformer classes. Where possible, it would be nice to have fewer if checks in snippets, which sometimes might mean making view models more abstract sometimes. For example, here we could put the name of the transport-specific gax dependency in the model, knowing that we just need one of them. (Not a strict requirement for this PR, it's just an illustration.)